### ISSUE TITLE
[no-shadow] false positive for enum member

### ISSUE THREAD

**Issue by SimonSchick** [typescript-eslint/typescript-eslint#4340]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

Before anyone asks, this doesn't look like the same issue as https://github.com/typescript-eslint/typescript-eslint/issues/4336 it doesn't seem like `ignoreTypeValueShadow` or any other option stop this from happening.

My understand is the enum members are generally not added to the current scope (thank god), unlike in other languages, I believe this to be a legit error.

**Repro**

<details>
<summary>Config</summary>

```JSON
{
  "root": true,
  "ignorePatterns": "webpack.config.js",
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/all",
    "plugin:react/all",
    "plugin:import/recommended",
    "plugin:import/typescript",
    "prettier"
  ],
  "parserOptions": {
    "project": [
      "./src/webApp/tsconfig.json",
      "./src/chrome/tsconfig.json",
      "./src/webApp/build/tsconfig.json"
    ]
  },
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "@typescript-eslint",
    "react-hooks",
    "react",
    "import",
    "prettier"
  ],
  "rules": {
    "arrow-body-style": "off",
    "camelcase": "off",
    "class-methods-use-this": "off",
    "eol-last": "off",
    "eqeqeq": ["error", "always"],
    "global-require": "off",
    "import/export": "off",
    "import/extensions": [
      "error",
      {
        "pattern": {
          "scss": "always"
        }
      }
    ],
    "import/no-dynamic-require": "error",
    "import/no-extraneous-dependencies": "error",
    "import/no-unresolved": "error",
    "import/no-unused-modules": [
      "error",
      {
        "unusedExports": true
      }
    ],
    "linebreak-style": "off",
    "max-len": ["error", 200], // covered by prettier but #yolo
    "no-console": "error",
    "no-floating-decimal": "error",
    "no-invalid-this": "error",
    "no-lonely-if": "error",
    "no-nested-ternary": "error",
    "no-underscore-dangle": "off",
    "no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "lodash",
            "message": "Only submodule imports are permitted."
          },
          {
            "name": "aws-sdk",
            "message": "Only submodule imports are permitted."
          }
        ]
      }
    ],
    "object-curly-newline": "off",
    "one-var-declaration-per-line": "error",
    "one-var": ["error", "never"],
    "padded-blocks": "off",
    "prefer-arrow-callback": "error",
    "prefer-const": "error",
    "prefer-destructuring": "error",
    "prettier/prettier": "error",
    "react-hooks/exhaustive-deps": "error",
    "react-hooks/rules-of-hooks": "error",
    "react/destructuring-assignment": "error",
    // It is often desirable to pass className
    "react/forbid-component-props": "off",
    "react/jsx-curly-brace-presence": ["error", "never"],
    "react/jsx-filename-extension": ["error", { "extensions": [".tsx"] }],
    "react/function-component-definition": [
      "error",
      {
        "namedComponents": "arrow-function",
        "unnamedComponents": "arrow-function"
      }
    ],
    "react/jsx-key": "error",
    "react/jsx-max-depth": [
      "error",
      {
        "max": 8
      }
    ],
    "react/jsx-no-bind": "error",
    // Reserved for future i18n use
    "react/jsx-no-literals": "off",
    // We use this relatively often hen passing through arbitray properties to chidren
    // and there isn't much of a downside using this with TS
    "react/jsx-props-no-spreading": "off",
    "react/jsx-sort-props": "off",
    "react/jsx-wrap-multilines": "error",
    "react/no-array-index-key": "error",
    // We often break out tiny helper components
    // that are highly specialised, it often
    // doesn't make sense to split them into different files
    "react/no-multi-comp": "off",
    "react/no-unstable-nested-components": "error",
    // generally not useful as TS covers undeclared props
    // this also causes issues w/ forwardRef etc.
    "react/prop-types": "off",
    // Not useful as often use `undefined` or optionals
    "react/require-default-props": "off",
    // We don't use functional components
    "react/sort-comp": "off",
    "no-else-return": [
      "error",
      {
        "allowElseIf": false
      }
    ],

    // Probably not turning these on
    // We deal with a lot of snake case due to API, this would be rather annoying
    "@typescript-eslint/naming-convention": "off",
    // This is kinda useless and prevents type aliasing.
    "@typescript-eslint/no-type-alias": "off",
    // This is a nice feature of TS, unsure why we would ban this
    "@typescript-eslint/no-parameter-properties": "off",
    // Waay too many false positives where the behavior is desired and concise
    "@typescript-eslint/strict-boolean-expressions": "off",

    // Style heavy
    "@typescript-eslint/lines-between-class-members": "off",
    "@typescript-eslint/prefer-enum-initializers": "off",
    "@typescript-eslint/padding-line-between-statements": [
      "error",
      {
        "prev": [
          "multiline-block-like",
          "multiline-const",
          "multiline-expression",
          "multiline-let",
          "multiline-var"
        ],
        "next": "*",
        "blankLine": "always"
      }
    ],

    // Potentially turning these on
    "@typescript-eslint/member-ordering": "off",
    "@typescript-eslint/prefer-readonly-parameter-types": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-magic-numbers": "off",
    "@typescript-eslint/consistent-type-imports": "off",
    "@typescript-eslint/sort-type-union-intersection-members": "off",
    "@typescript-eslint/no-shadow": [
      "error",
      {
        "ignoreTypeValueShadow": true
      }
    ],
    "@typescript-eslint/no-use-before-define": "off",
    "@typescript-eslint/method-signature-style": "off",

    // Turning these on progressively
    "@typescript-eslint/no-non-null-assertion": "off",
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/explicit-module-boundary-types": "error",
    "@typescript-eslint/no-unnecessary-condition": [
      "error",
      {
        "allowConstantLoopConditions": true
      }
    ],
    "@typescript-eslint/no-unsafe-assignment": "error",
    "@typescript-eslint/no-unsafe-call": "error",
    "@typescript-eslint/no-unsafe-member-access": "error",
    "@typescript-eslint/no-unsafe-return": "error",
    "@typescript-eslint/no-unnecessary-type-assertion": "error",
    "@typescript-eslint/restrict-template-expressions": "error",
    "@typescript-eslint/no-unsafe-argument": "error",
    "@typescript-eslint/restrict-plus-operands": "error"
  },
  "settings": {
    "react": {
      "version": "17.0.2"
    },
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"]
      }
    }
  }
}
```
</details>

```TS
const MyVar = 123;

enum Test {
  MyVar = 1,
}
```

```json
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "allowUnreachableCode": false,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "inlineSources": true,
    "isolatedModules": true,
    "jsx": "react",
    "lib": ["dom", "dom.iterable", "esnext"],
    "module": "esnext",
    "moduleResolution": "node",
    "noFallthroughCasesInSwitch": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "sourceMap": true,
    "sourceRoot": "/",
    "strict": true,
    "target": "ES2018",
    "types": ["webpack-env"]
  },
  "include": ["src"]
}
```

**Expected Result**

No error generated.

**Actual Result**
`'MyVar' is already declared in the upper scope.eslint@typescript-eslint/no-shadow`

**Additional Info**

Nothing really.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.8.0` |
| `@typescript-eslint/parser`        | `5.8.0` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.5.0` |
| `node`                             | `17.0.1` |
| `all other eslint packages`   | `latest` |

**Comment by SimonSchick** [comment]
Closing this, didn't find https://github.com/typescript-eslint/typescript-eslint/issues/2721#issuecomment-718804636 on my first search.
